zeus/optimizer/batch_size/server/optimizer.py

```diff
@@ -0,0 +1,310 @@
+from __future__ import annotations
+
+from collections import defaultdict
+from uuid import UUID
+from fastapi import HTTPException
+
+import numpy as np
+from zeus.optimizer import batch_size
+
+from zeus.optimizer.batch_size.server.models import (
+    JobSpec,
+    ReportResponse,
+    TrainingResult,
+)
+from zeus.optimizer.batch_size.server.mab import GaussianTS
+from zeus.optimizer.batch_size.server.explorer import PruningExploreManager
+from zeus.util.metric import zeus_cost
+
+# from dbapis import DBAPI
+
+GLOBAL_ZEUS_SERVER: ZeusBatchSizeOptimizer | None = None
+
+"""
+TODO: ADD EARLY STOPPING SOMEHOW?. If cost is too much, stop training 
+"""
+
+
+class ZeusBatchSizeOptimizer:
+    """A singleton class that manages training jobs. Why singleton? Not sure yet.
```

Comment:
(0) It doesn't really have to be a singleton, as long as you have a way to retrieve and set job states. In the current design, this class is a singleton and manages the job state of **every** job, which is why all the dictionaries are keyed with job UUIDs.
(1) Yeah, all the dictionaries set as instance attributes (e.g., `self.mabs`)
(2) Eventually all these will go to the DB, and this class will be the class that interfaces with the DB. Basically every access to `self.mabs`, `self.history`, etc will be a DB query done through an ORM. However, this still doesn't force you to make this class a singleton.

Reply:
Agreed!